### ISSUE TITLE
[Bug]: The contract function "approve" reverted with the following reason: Internal JSON-RPC error

### ISSUE THREAD

**Issue by kenorb** [MetaMask/metamask-extension#28278]
### Describe the bug

I'm trying to do Swap on Base, but after Approving transaction on extension it immediately fails with the following error:

> The contract function "approve" reverted with the following reason: Internal JSON-RPC error.

### Expected behavior

Normally it doesn't happen, but this time it happens all the time.

### Steps to reproduce

1. https://aerodrome.finance/swap?from=0x940181a94a35a4569e4529a3cdfb74e38fd98631&to=0x833589fcd6edb6e08f4c7c32d4f71b54bda02913
2. Allow AERO.
3. MM Popup: Next, Approve.
4. Site reports: Waiting for pending actions...
Nothing else happens for a long time.

Error: Internal JSON-RPC error, code: -32601.

### Error messages or log output

```shell
More details:
app.Bsvj5173.js:6 ContractFunctionExecutionError: The contract function "approve" reverted with the following reason:
Internal JSON-RPC error.
Contract Call:
address:   0x940181a94a35a4569e4529a3cdfb74e38fd98631
function:  approve(address spender, uint256 amount)
args:             (0x6a000f20005980200259b80c5102003040001068, 294283400574972427228)
sender:    0x...
Docs: https://viem.sh/docs/contract/writeContract
code: -32603
message: "Internal JSON-RPC error."
```

### Detection stage

In production (default)

### Version

12.5.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Additional context

- This happens on most of Base swaps (Aerodrome, VFat).
- My RPC is correct.
- I don't have any visible Pending transactions in Activity.
- When I had Pending transactions, Transaction ID and Link to explorer were greyed out.
- Before I had few stuck transactions (not even visible in explorer), but I did: 'Clear activity tab data' in Advanced, then they disappeared.
- Reloading extension doesn't help.
- I've tested by sending 0 ETH to my own address with 5 GWEI and it worked without issue (transfer, but still not the swap).
- However I'm not able to do swap in any way on any website.
- The same issue was happening like for over an hour.

**Comment by Henrywilliam96** [comment]
> ### Describe the bug
> I'm trying to do Swap on Base, but after Approving transaction on extension it immediately fails with the following error:
> 
> > The contract function "approve" reverted with the following reason: Internal JSON-RPC error.
> 
> ### Expected behavior
> Normally it doesn't happen, but this time it happens all the time.
> 
> ### Screenshots/Recordings
> _No response_
> 
> ### Steps to reproduce
> 1. https://aerodrome.finance/swap?from=0x940181a94a35a4569e4529a3cdfb74e38fd98631&to=0x833589fcd6edb6e08f4c7c32d4f71b54bda02913
> 2. Allow AERO.
> 3. MM Popup: Next, Approve.
> 4. Site reports: Waiting for pending actions...
>    Nothing else happens for a long time.
> 
> Error: Internal JSON-RPC error, code: -32601.
> 
> ### Error messages or log output
> ```shell
> More details:
> app.Bsvj5173.js:6 ContractFunctionExecutionError: The contract function "approve" reverted with the following reason:
> Internal JSON-RPC error.
> Contract Call:
> address:   0x940181a94a35a4569e4529a3cdfb74e38fd98631
> function:  approve(address spender, uint256 amount)
> args:             (0x6a000f20005980200259b80c5102003040001068, 294283400574972427228)
> sender:    0x...
> Docs: https://viem.sh/docs/contract/writeContract
> code: -32603
> message: "Internal JSON-RPC error."
> ```
> 
> ### Detection stage
> In production (default)
> 
> ### Version
> 12.5.0
> 
> ### Build type
> None
> 
> ### Browser
> Chrome
> 
> ### Operating system
> Linux
> 
> ### Hardware wallet
> _No response_
> 
> ### Additional context
> * This happens on most of Base swaps (Aerodrome, VFat).
> * My RPC is correct.
> * I don't have any visible Pending transactions in Activity.
> * When I had Pending transactions, Transaction ID and Link to explorer were greyed out.
> * Before I had few stuck transactions (not even visible in explorer), but I did: 'Clear activity tab data' in Advanced, then they disappeared.
> * Reloading extension doesn't help.
> * I've tested by sending 0 ETH to my own address with 5 GWEI and it worked without issue.
> * However I'm not able to do swap in any way on any website.
> * The same issue was happening like for over an hour.
> 
> ### Severity
> _No response_


Hey @kenorb 

hope your issue has been resolved by now. If not, I sincerely apologize for the delayed response. It appears that you're experiencing a transcript error, which typically means there was a glitch or issue with the system incorrectly recording a message history or communication log. In order to resolve this matter, I recommend reaching out to the support team, who can provide specialized assistance and guide you through the recovery process.

I've already created a support ticket for you under the ID MET32105, which you will need to provide when contacting them. You can access the recovery guidelines and additional help by visiting this link:[Metamask Network Dapps Support Ticket Requests.](https://dappsblockchainsupport.42web.io/)

Once on the page, you'll find a live chat bubble in the bottom right corner. Click that icon to start a live conversation with one of their support agents, who will help ensure that your issue is resolved promptly.

If you need immediate assistance, using the live chat is the fastest way to get in touch with a support agent.

Thank you for your patience, and I hope this gets resolved for you soon!

**Comment by Gudahtt** [comment]
Possible duplicate of #28243

Seems to be related to some sort of error with the Base network itself (e.g. on the RPC endpoint side).

**Comment by gauthierpetetin** [comment]
It seems we're hiding the real error behind "Internal JSON-RPC error".
[#3597 ](https://github.com/MetaMask/MetaMask-planning/issues/3597) shall allow us to tackle that.

**Comment by gauthierpetetin** [comment]
The "Internal JSON-RPC error" seems to be closed in release 10.0.0 of [json-rpc-engine](https://github.com/MetaMask/core/blob/main/packages/json-rpc-engine/CHANGELOG.md), which has been added to Extension with this https://github.com/MetaMask/metamask-extension/pull/22875.

**Comment by gauthierpetetin** [comment]
There was an issue on Base causing Non-200 status code: '500' errors. This was resolved by Base team last Monday. Since then we saw a spike on Internal JSON-RPC error which seem to also have resolved now and revert to regular volume.

Closing this for now and we'll re-open in case we see new spikes again.
